README.rst

```diff
@@ -17,25 +17,63 @@ The widget takes the following parameters:
  - override: drops all global config and the base query if a list is passed to the widget. All types need to be added to be selectable.
 
 
-**IMPORTANT NOTE:**
-Currently this widget drops the SourceBinder concept, which has huge impact on the usability.
+ContextSourceBinder
+-------------------
+
+With a `RelationeChoice` or `RelationList` of `RelationChoice` a source can be configured along with the field.
+The ContextSourceBinder makes sure, the right content will be referenced.
+
+The default filter method, therefore only checks for the correct type:
+
+.. code:: python
+
+    def default_filter(source, value):
+        return value.portal_type in get_selectable_types_by_source(source)
+
+Feel free to add your own filter method as source parameter in your field.
+Example:
+
+.. code:: python
+
+    def custom_filter(source, value):
+        return bool(..)
+
+    ...
+
+    directives.widget(realtionchoice_restricted_title=ReferenceWidgetFactory)
+    realtionchoice_restricted_title = RelationChoice(
+        title=_(u'Related Choice Restricted Title'),
+        source=ReferenceObjSourceBinder(
+            selectable_function=custom_filter),
+        default=None,
+        required=False,
+    )
+
+The `filter` takes two parameter the actual source object and a value, which is the content object.
+
+Only `ReferenceObjSourceBinder` are supported. The SourceBinder takes the following parameters:
+
+- selectable: These Types are selectable
+- nonselectable: These Types are not selectable
+- allow_nonsearched_types: If this is set to true all the types will be traversable and selectable.
+- override: drops all global config and the base query if a list is passed to the widget. All types need to be added to be selectable.
```

Comment:
Actually I don't know... thats from @tschanzt 
Ahh I see... I the override flag is False (by default) the `selectable` types are added to the list of `global` selectable types  

**Global** means: Combined list of types from the registry and `types_not_searched`, etc.